File: o2o/trunk/web/src/main/webapp/js/HuiyuanAdv.js
/*
 * Copyright (c) 2009-2016 SHENZHEN Eternal Dynasty Technology Co.,Ltd.
 * All rights reserved.
 *  
 * This file contains valuable properties of  SHENZHEN Eternal Dynasty 
 * Technology Co.,Ltd.,  embodying  substantial  creative efforts  and 
 * confidential information, ideas and expressions.    No part of this 
 * file may be reproduced or distributed in any form or by  any  means,
 * or stored in a data base or a retrieval system,  without  the prior 
 * written permission  of  SHENZHEN Eternal Dynasty Technology Co.,Ltd.
 *
 */

$(function (){
	$("#left").click(function(){
		cout--;
		if(cout<1){
			alert("已经在最左边了！");
			cout++;
		}else{
			goleft(cout);
		}
	});
	
$("#right").click(function(){
	if($(".xiazhan_xiak").length<4){
		alert("已经到最右边了！");
	}else{
		cout++;
		goright(cout);
		}
	});
	
});

function  goleft(num){
	$.ajax({
			  type: 'POST',
			  url: servicePath+"/widget.html",
			  data: "widget=huiyuanAdv&left="+num,
			  dataType: "html",
			  success: function(data){
			  	$(".xiazhan_kuai").html(data)
				}
			});	
}

function  goright(num){

$.ajax({
			  type: 'POST',
			  url: servicePath+"/widget.html",
			  data: "widget=huiyuanAdv&right="+num,
			  dataType: "html",
			  success: function(data){
				$(".xiazhan_kuai").html(data)
				}
			});	

}

File: o2o/trunk/java/o2o-web/src/main/java/cn/com/dyninfo/o2o/furniture/web/member/widget/HuiyuanAdv.java
/*
 * Copyright (c) 2009-2016 SHENZHEN Eternal Dynasty Technology Co.,Ltd.
 * All rights reserved.
 *
 * This file contains valuable properties of  SHENZHEN Eternal Dynasty
 * Technology Co.,Ltd.,  embodying  substantial  creative efforts  and
 * confidential information, ideas and expressions.    No part of this
 * file may be reproduced or distributed in any form or by  any  means,
 * or stored in a data base or a retrieval system,  without  the prior
 * written permission  of  SHENZHEN Eternal Dynasty Technology Co.,Ltd.
 *
 */

package cn.com.dyninfo.o2o.furniture.web.member.widget;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

import javax.annotation.Resource;
import javax.servlet.http.HttpSession;

import org.springframework.stereotype.Component;

import cn.com.dyninfo.o2o.furniture.util.PageInfo;

import cn.com.dyninfo.o2o.furniture.web.framework.context.Context;
import cn.com.dyninfo.o2o.old.service.PagModInGoodsService;
import cn.com.dyninfo.o2o.old.model.HuiyuanInfo;
import cn.com.dyninfo.o2o.old.model.Order;
import cn.com.dyninfo.o2o.old.model.OrderProduct;
import cn.com.dyninfo.o2o.old.service.OrderService;

@Component("huiyuanAdv")
public class HuiyuanAdv extends AbstractMemberWidget {
	@Resource
	private PagModInGoodsService pagModInGoodsService;
	@Resource
	private OrderService orderService;
	
	@Override
	public void execute(Map pamtr) {
		PageInfo page=new PageInfo();
		String left=(String)pamtr.get("left");
		String right=(String)pamtr.get("right");
	
		if(left!=null){
			page.setPageNo(Integer.parseInt(left));
		}else if(right!=null){
			page.setPageNo(Integer.parseInt(right));
		}else{
			page.setPageNo(1);
		}
		page.setPageSize(4);
		HttpSession session=HttpRequest.getSession();
		HuiyuanInfo huiyuan = (HuiyuanInfo) session.getAttribute(Context.SESSION_MEMBER);
		List list=orderService.getListByWhere(new StringBuffer("and n.huiyuan.huiYuan_id="+huiyuan.getHuiYuan_id()+"order by creatTime desc"));
		if(list.size()>0){
			Order  order  =(Order)list.get(0);;
			OrderProduct orderproductlist=(OrderProduct) order.getOrderProductList().get(0);
			int	goodSort =orderproductlist.getProduct().getGoodsSort().getGoodsSort_id();
			List<Map> datalist= pagModInGoodsService.getGoodBySlqCount(goodSort+"", "", page, "", "41" );
			this.putData("data", datalist);
		}else{
			this.putData("data", new ArrayList());
		}
	}

	
}
